cli/azd/pkg/infra/provisioning/terraform/terraform_provider.go

```diff
@@ -0,0 +1,559 @@
+	if err != nil {
+		return nil, fmt.Errorf("reading deployment output failed: %s", runResult.Stderr)
+	}
+
+	var outputMap map[string]map[string]interface{}
+	if err := json.Unmarshal([]byte(runResult.Stdout), &outputMap); err != nil {
+		return nil, err
+	}
+
+	outputParameters := make(map[string]OutputParameter)
+	for k, v := range outputMap {
+		outputParameters[k] = OutputParameter{
+			Type:  fmt.Sprint(v["type"]),
+			Value: v["value"],
+		}
+	}
+	return outputParameters, nil
+}
+
+// Creates the deployment object from the specified module path
+func (t *TerraformProvider) createDeployment(ctx context.Context, modulePath string) (*Deployment, error) {
+	templateParameters := make(map[string]InputParameter)
+
+	//build the template parameters.
+	parameters := make(map[string]any)
+	parametersFilePath := t.parametersFilePath()
+
+	// check if the file does not exist to create it --> for shared env scenario
+	log.Printf("Reading parameters template file from: %s", parametersFilePath)
+	if _, err := os.Stat(parametersFilePath); err != nil {
+		err = t.createParametersFile()
+		if err != nil {
+			return nil, fmt.Errorf("creating parameters file: %w", err)
+		}
+	}
+
+	parametersBytes, err := os.ReadFile(parametersFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("reading parameter file template: %w", err)
+	}
+
+	if err := json.Unmarshal(parametersBytes, &parameters); err != nil {
+		return nil, fmt.Errorf("error unmarshalling template parameters: %w", err)
+	}
+
+	for key, param := range parameters {
+		templateParameters[key] = InputParameter{
+			Type:  key,
+			Value: param,
+		}
+	}
+
+	template := Deployment{
+		Parameters: templateParameters,
+	}
+
+	return &template, nil
+}
+
+// Copies the Terraform parameters file from the project template into the .azure environment folder
+func (t *TerraformProvider) createParametersFile() error {
+	// Copy the parameter template file to the environment working directory and do substitutions.
+	parametersTemplateFilePath := t.parametersTemplateFilePath()
+
+	log.Printf("Reading parameters template file from: %s", parametersTemplateFilePath)
+	parametersBytes, err := os.ReadFile(parametersTemplateFilePath)
+	if err != nil {
+		return fmt.Errorf("reading parameter file template: %w", err)
+	}
+	replaced, err := envsubst.Eval(string(parametersBytes), func(name string) string {
+		if val, has := t.env.Values[name]; has {
+			return val
+		}
+		return os.Getenv(name)
+	})
+
+	if err != nil {
+		return fmt.Errorf("substituting parameter file: %w", err)
+	}
+
+	parametersFilePath := t.parametersFilePath()
+	writeDir := filepath.Dir(parametersFilePath)
+	if err := os.MkdirAll(writeDir, 0755); err != nil {
+		return fmt.Errorf("creating directory structure: %w", err)
+	}
+
+	log.Printf("Writing parameters file to: %s", parametersFilePath)
+	err = os.WriteFile(parametersFilePath, []byte(replaced), 0644)
+	if err != nil {
+		return fmt.Errorf("writing parameter file: %w", err)
+	}
+
+	return nil
```

Comment:
It would be great if we can consolidate the evaluation and replacement of the files between bicep & terraform into a separate package that both can leverage.

Reply:
done